### ISSUE TITLE
`single-commit` doesn't delete old commits

### ISSUE THREAD

**Issue by mayank99** [JamesIves/github-pages-deploy-action#756]
_**TL;DR:** Using single-commit doesn't erase git history, it just seems to "hide" old commits (so there is only one "visible" commit in deployed branch)._

## Describe the bug

I noticed this when I was looking at my "github home feed" ([github.com](https://github.com)). I have a workflow that uses single-commit, and yet I saw multiple commits in the feed from github-actions bot (the git user I set in my workflow).

![image](https://user-images.githubusercontent.com/9084735/121073794-00157380-c7a1-11eb-8643-a682b2bb9085.png)

I scrolled down some more and was able to access commits from weeks ago! (Makes me wonder if this could cause problems with storage, as the deployments add up over time).

## Reproduction Steps

1. Set `single-commit: true` in workflow.
2. Run workflow, get commit link (from deployed branch).
3. Run workflow again, visit old commit link.

## Logs

Logs have nothing special. Should be reproducible in any standard workflow.

```yml
- name: Host stuff on gh pages
  uses: JamesIves/github-pages-deploy-action@4.1.3
  with:
    branch: gh-pages
    folder: ${{ github.workspace }}/stuff
    target-folder: stuff
    single-commit: true
    clean: true
    git-config-name: github-actions[bot]
    git-config-email: github-actions[bot]@users.noreply.github.com
```

<details>
<summary>Logs</summary>

```
Run JamesIves/github-pages-deploy-action@4.1.3
  with:
    branch: gh-pages
    folder: /home/runner/work/RepoName/RepoName/stuff
    target-folder: stuff
    single-commit: true
    git-config-name: github-actions[bot]
    git-config-email: github-actions[bot]@users.noreply.github.com
    token: ***
    clean: true

    GitHub Pages Deploy Action 🚀

    🚀 Getting Started Guide: https://github.com/marketplace/actions/deploy-to-github-pages
    ❓ Discussions / Q&A: https://github.com/JamesIves/github-pages-deploy-action/discussions
    🔧 Report a Bug: https://github.com/JamesIves/github-pages-deploy-action/issues

    📣 Maintained by James Ives: https://jamesiv.es
    💖 Support: https://github.com/sponsors/JamesIves
Checking configuration and starting deployment… 🚦
Deploying using Deploy Token… 🔑
Configuring git…
/usr/bin/git config user.name github-actions[bot]
/usr/bin/git config user.email github-actions[bot]@users.noreply.github.com
/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
/usr/bin/git remote rm origin
/usr/bin/git remote add origin ***github.com/UserName/RepoName.git
Git configured… 🔧
Starting to commit changes…
/usr/bin/git ls-remote --heads ***github.com/UserName/RepoName.git refs/heads/gh-pages
c9457e7e0232bb7cf28db2142302b2d4697d5ff9	refs/heads/gh-pages
Creating worktree…
/usr/bin/git fetch --no-recurse-submodules --depth=1 origin gh-pages
From https://github.com/UserName/RepoName
 * branch            gh-pages   -> FETCH_HEAD
 * [new branch]      gh-pages   -> origin/gh-pages
/usr/bin/git worktree add --no-checkout --detach github-pages-deploy-action-temp-deployment-folder
Preparing worktree (detached HEAD 83fbdbc)
/usr/bin/git checkout --orphan gh-pages origin/gh-pages
Previous HEAD position was 83fbdbc Merge 99315fd796099a8352a430e376cb8a089e21a394 into 838f20db9dad4dbe8d30c77f4fa696db0de53f8a
Switched to a new branch 'gh-pages'
Creating target folder if it doesn't already exist… 📌
/usr/bin/rsync -q -av --checksum --progress /home/runner/work/RepoName/RepoName/stuff/. github-pages-deploy-action-temp-deployment-folder/97 --delete --exclude CNAME --exclude .nojekyll --exclude .ssh --exclude .git --exclude .github
Checking if there are files to commit…
/usr/bin/git add --all .
/usr/bin/git checkout -b github-pages-deploy-action/yo2hn04mf
Switched to a new branch 'github-pages-deploy-action/yo2hn04mf'
/usr/bin/git commit -m Deploying to gh-pages from @ UserName/RepoName@83fbdbc20ec2835b897b485e21d4dddf68587ab0 🚀 --quiet --no-verify
/usr/bin/git push --force ***github.com/UserName/RepoName.git github-pages-deploy-action/yo2hn04mf:gh-pages
To https://github.com/UserName/RepoName.git
 + c9457e7...02d33dd github-pages-deploy-action/yo2hn04mf -> gh-pages (forced update)
Changes committed to the gh-pages branch… 📦
Running post deployment cleanup jobs… 🗑️
/usr/bin/git checkout -B github-pages-deploy-action/yo2hn04mf
Reset branch 'github-pages-deploy-action/yo2hn04mf'
/usr/bin/git worktree remove github-pages-deploy-action-temp-deployment-folder --force
Completed deployment successfully! ✅
```
</details>

## Additional Comments

Additionally, I saw that github-actions was the top contributor in the repo (despite only having one "visible" commit)! 😄 
![image](https://user-images.githubusercontent.com/9084735/121074435-d872db00-c7a1-11eb-804f-4daa6707a395.png)

**Comment by JamesIves** [comment]
Thanks for the report,

The intention of the single-commit option is to squash everything on the branch so it appears as if there's one commit, this was a common request as a number of projects already perform such a process on their deployment branch. 

If the commits are still visible due to some artifacts that aren't discarded by GitHub then I think an additional process outside of this action should take care of that. The single-commit option already feels destructive as it is, so ensuring permanent deletion of the commits it squashes is something I'm going to deem as out of scope.

Please let me know if you feel differently.
